backend/wayland/backend.c

```diff
@@ -436,42 +442,27 @@ struct wlr_backend *wlr_wl_backend_create(struct wl_display *display,
 	}
 	wl_event_source_check(wl->remote_display_src);
 
-	wl->drm_fd = -1;
 	if (wl->drm_render_name != NULL) {
 		wlr_log(WLR_DEBUG, "Opening DRM render node %s", wl->drm_render_name);
 		wl->drm_fd = open(wl->drm_render_name, O_RDWR | O_NONBLOCK | O_CLOEXEC);
 		if (wl->drm_fd < 0) {
 			wlr_log_errno(WLR_ERROR, "Failed to open DRM render node %s",
 				wl->drm_render_name);
-			goto error_remote_display_src;
-		}
-
-		int drm_fd = fcntl(wl->drm_fd, F_DUPFD_CLOEXEC, 0);
-		if (drm_fd < 0) {
-			wlr_log(WLR_ERROR, "fcntl(F_DUPFD_CLOEXEC) failed");
-			goto error_drm_fd;
-		}
-
-		struct wlr_gbm_allocator *gbm_alloc = wlr_gbm_allocator_create(drm_fd);
-		if (gbm_alloc == NULL) {
-			wlr_log(WLR_ERROR, "Failed to create GBM allocator");
-			close(drm_fd);
-			goto error_drm_fd;
+			wl_event_source_remove(wl->remote_display_src);
```

Comment:
I think this `if` block should returns in a fatal error.

Reply:
Shouldn't we fall back to software rendering if the drm render node fails to open?

Reply:
If `wl->drm_render_name` is not NULL, then the parent compositor has advertised a DRM node. If a DRM node has been advertised but we aren't able to open it, it means that a GPU is available but something really bad happened. I'd prefer to complain loudly than to have a fallback. Users won't expect wlroots to use software rendering if a GPU is available. Users will notice the loud complaining, but won't notice the fallback.